osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -99,20 +88,13 @@ public WorkingBeatmap Beatmap
                 if (beatmap == value) return;
 
                 beatmap = value;
-                cancellationSource?.Cancel();
-                cancellationSource = new CancellationTokenSource();
-
-                beatmapDifficulty?.UnbindAll();
-                beatmapDifficulty = difficultyCache.GetBindableDifficulty(beatmap.BeatmapInfo, cancellationSource.Token);
-                beatmapDifficulty.BindValueChanged(_ => updateDisplay());
```

Comment:
If you can bring back this logic and make things work (hopefully reducing the lines change in this PR by 50% in the process) that would be a good starting point.